Add Nginx Reverse proxy

robertcsakany opened the pull request:
Add support of nginx reverse proxy. It tracking all container where VIRTUAL_HOST env is defined, 
automatically generate nginx proxy config for it.

As described in https://github.com/nginx-proxy/nginx-proxy. We use separate containers.
The HTTPS implementation as documented here: https://medium.com/@francoisromain/host-multiple-websites-with-https-inside-docker-containers-on-a-single-server-18467484ab95

By default the xip.io is used to be able make subdomains for IP address. So for example the nodered service can be accessed:

nodered.X.X.X.X.xip.io 

where X.X.X.X is the IP address of IOTstack. 

Any other domain can be used, on that case please replace VIRTUAL_HOST env variable of the given instance with the coresponding value.

HTTPS can be used, but xip.io method is not suitable for that, so as the links describes any container can be exposed with the definition of HTTPS proto, but some domain have to be defined.

Paraphraser committed:
Adds a Gitea container to IOTstack

Gitea is a lightweight implementation of a Git server. Makers who are
already running a local IOTstack server may prefer to commit their work
to a local repository running on their Raspberry Pi, rather than
GitHub/GitLab/etc. Implementing Gitea under the IOTStack umbrella
inherits the existing IOTstack backup mechanisms for free.

Candidate page for the wiki at
https://www.dropbox.com/s/onqndgyamo2h74i/Gitea.md.zip?dl=1

Tomasmerca committed:
Added filter.acl file to mosquitto/config

Route filter.acl to add the option in mosquitto/mosquitto.conf of filtering wich topics can be read or writen by defined users.

Tomasmerca committed:
Created empty file whit example filter

Added the file where the user creates filters.
The example filters don´t affect usage in any way.

Tomasmerca committed:
Added options for password and filter config

RavHilton committed:
Update menu.sh

Version "2" of the docker-compose.yml cannot use "Healthcheck" instructions.  Only 2.1+ can do this.  I propose we upgrade the version to at least 3 so we can incorporate HealthCheck commands and any other new features as appropriate.  We're running the latest version of docker - the latest version of the docker-compose.yml is 3.7 - unless there's a really, really good reason to use an older format of the docker-compose.yml, can we please update it to at least 3.0, preferably 3.7?

gmacario committed:
menu.sh: Fix small typo in msg "Restart Required"

s/restart you device/restart your device/

robertcsakany committed:
Add Nginx Reverse proxy

Slyke commented:
I have submitted this PR on your behalf over at the fork: https://github.com/SensorsIot/IOTstack/pull/1, If you would like to have your name to the commits, please submit the PR there and let me know so I can remove the PR I created on your behalf.

Willem-Dekker committed:
first import

Willem-Dekker committed:
added index.md

Willem-Dekker committed:
Updated styling of existing pages

Willem-Dekker committed:
Updated styling of existing pages

SensorsIot committed:
Merge pull request #20 from Willem-Dekker/mkdocks

Added Mkdocks action and github pages intergration for the documentation.

SensorsIot committed:
Update README.md

SensorsIot committed:
Update README.md

Willem-Dekker committed:
Merge pull request #16 from SensorsIot/pr/161

menu.sh: Fix small typo in msg "Restart Required"

Commit:
Changed each menu item to be on a separate line to make merges easier.

Commit:
Merge branch 'armhf_keys_menu_lines' into pr/126

Commit:
Changed docker-compose yml file version to 3.6.

Willem-Dekker committed:
Merge pull request #26 from SensorsIot/armhf_keys_menu_lines

Changed each menu item to be on a separate line to make merges easier.

Slyke committed:
Update README.md

Updated readme with quick install instructions so people don't have to dig for it in the documentation.

Willem-Dekker committed:
Merge pull request #7 from SensorsIot/pr/139

Update menu.sh

Willem-Dekker committed:
Merge pull request #3 from SensorsIot/pr/126

Adds a Gitea container to IOTstack

Willem-Dekker committed:
Merge pull request #27 from SensorsIot/readme-update

Update README.md

Willem-Dekker committed:
Merge pull request #4 from SensorsIot/pr/130

Mosquitto changes update to security

robertcsakany committed:
Merge branch 'master' into feature/AddNginxProxy

enriquedelpino commented:
> robertcsakany

Hi robert,

I've tried to bring your changes to my personal fork of IOTstack and have them running. After I got the script to generate the entries for the containers you added, I don't see the nginx configuration being generated anywhere, so I was wondering if there is something that needs to be done in order to generate the config correctly.

Paraphraser commented:
Hi @enriquedelpino 

I've been watching your [PR638](https://github.com/SensorsIot/IOTstack/pull/638) on the SensorsIot/IOTstack repo. I'm going to make some more comments there in a bit.

I'm assuming the reason why you've posted this message on gcgarner/IOTstack is because @robertcsakany started working on an nginx proxy here on gcgarner before this repo (gcgarner) was forked to SensorsIot after Graham Garner stopped responding.

> There's more background at [This Project is Dormant](https://github.com/gcgarner/IOTstack/issues/194).

PR174 on gcgarner (this PR) was inherited by SensorsIot as [PR21](https://github.com/SensorsIot/IOTstack/pull/21) but it is in the same state of having conflicts that must be resolved.

The point I'm trying to make is that this repo (gcgarner) is effectively abandoned so it is highly unlikely that anything you do here will get anywhere. On the other hand, because PR21 on SensorsIot hasn't moved in a very long time, it's probably reasonable to conclude that it is also unlikely to go anywhere.

I'll now move to your PR638 and continue my comments there.
